src/main/java/org/fcrepo/migration/validator/Driver.java

```diff
@@ -37,33 +39,63 @@
 
     private static final Logger LOGGER = getLogger(Driver.class);
 
-    private enum F3SourceTypes {
-        AKUBRA, LEGACY, EXPORTED;
-
-        static F3SourceTypes toType(final String v) {
-            return valueOf(v.toUpperCase());
-        }
-    }
-
-    @CommandLine.Option(names = {"--source-type", "-t"}, required = true, order = 1,
+    @CommandLine.Option(names = {"--source-type", "-s"}, required = true, order = 1,
             description = "Fedora 3 source type. Choices: akubra | legacy | exported")
     private F3SourceTypes f3SourceType;
 
     @CommandLine.Option(names = {"--datastreams-dir", "-d"}, order = 2,
-            description = "Directory containing Fedora 3 datastreams (used with --source-type 'akubra' or 'legacy')")
+            description = "Directory containing Fedora 3 datastreams " +
+                    "(used with --source-type 'akubra' or 'legacy')")
     private File f3DatastreamsDir;
 
     @CommandLine.Option(names = {"--objects-dir", "-o"}, order = 3,
             description = "Directory containing Fedora 3 objects (used with --source-type 'akubra' or 'legacy')")
     private File f3ObjectsDir;
 
+    @CommandLine.Option(names = {"--exported-dir", "-e"}, order = 4,
+            description = "Directory containing Fedora 3 export (used with --source-type 'exported')")
+    private File f3ExportedDir;
+
+    @CommandLine.Option(names = {"--f3hostname", "-f"},
+            defaultValue = "fedora.info", showDefaultValue = ALWAYS, order = 5,
+            description = "Hostname of Fedora 3, used for replacing placeholder in 'E' and 'R' datastream URLs")
+    private String f3hostname;
+
+    @CommandLine.Option(names = {"--index-dir", "-i"}, order = 6,
+            description = "Directory where cached index of datastreams (will reuse index if already exists)")
+    private File indexDir;
+
+    @CommandLine.Option(names = {"--threads", "-t"}, order = 7,
+            description = "The number of threads for parallel processing. Default 5", defaultValue = "5")
+    private int threadCount;
+
+
     @CommandLine.Option(names = {"--debug"}, order = 30, description = "Enables debug logging")
     private boolean debug;
 
     @Override
     public Integer call() throws Exception {
-
-        return 0;
+        final var context = new AnnotationConfigApplicationContext("org.fcrepo.migration.validator");
```

Comment:
Maybe open the `context` in the `try` clause... then remove the explicit `close` block below.

Reply:
done.